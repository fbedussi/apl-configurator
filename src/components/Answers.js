import React from 'react';

const Answers = ({id, answers, setAnswer}) => {
    if (!answers) {
        return null;
    }

    return <fieldset className="answers">
            {answers.map(answer => {
                    var key = Object.keys(answer)[0];
                    var checked = false;
                    
                    return <label key={key} className="answerWrapper">
                        {answer[key]}
                        <input type="radio" name="answer" value={key} id={id + '_' + key} checked={checked} onClick={(e) => setAnswer(key)}/>
                    </label>;
                })
            }
    </fieldset>
}

export default Answers;