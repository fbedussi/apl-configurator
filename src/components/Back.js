import React from 'react';

const Back = ({ show, label, clickHandler }) => {
    if (!show) {
        return null;
    }

    return <div className="backWrapper">
            <button className="backBtn" onClick={clickHandler}>
            <span className="button-icon">
                <svg xmlns="http://www.w3.org/2000/svg" id="svg2" viewBox="0 0 29.57144 29.57144" 
                    height="29.571" width="29.571">
                    <g transform="translate(153.57 -600.29)" id="layer1" strokeLinecap="round">
                        <circle r="14.286" cy="615.076" cx="-138.786" id="circle" fill="gray" strokeWidth="0" />
                        <path id="arrow" d="M-140.283 606.237c-.61.017-1.19.272-1.617.71l-6.785 6.788c-1.855 1.676 0 3.322 0 3.322l7.142 7.143c2.215 2.293 5.617-1.107 3.325-3.323l-2.92-2.922h9.05c2.936 0 2.936-4.406 0-4.406h-9.765l3.277-3.278c1.538-1.494.437-4.1-1.707-4.035z" />
                    </g>
                </svg>
            </span>
            <span className="button-label">{label}</span>
        </button>
    </div>
}

export default Back;