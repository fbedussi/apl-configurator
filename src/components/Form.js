import React from 'react';
import { connect } from 'react-redux';

import { getShowForm } from '../selectors';
import { requestSent } from '../actions';

const mapStateToProps = (state) => ({
    showForm: getShowForm(state)
});

const mapDispatchToProps = (dispatch) => ({
    requestSent: () => dispatch(requestSent())
})

class Form extends React.Component {
    handleSubmit(e) {
        console.log(e);
        var form = e.target;
        var data = new FormData(form);
        fetch('send_form.php', {
            method: 'post',
            body: data
        })
        .then(response => {
            console.log(response);
            this.props.requestSent();
        })
        .catch(err => {
            console.log(err);
        })
    }

    render() {
        return <form onSubmit={(e) => this.handleSubmit(e)}
            className={`quotation-form ${this.props.showForm? 'show' : 'hide' }`}
            name="contactForm"
        >
            <label className="quotation-field" htmlFor="name">
                Nome
            </label>
                <input type="text" id="name" name="name"/>
            <label className="quotation-field">
                Cognome
                <input type="text" id="surname" name="surname"/>
            </label>
            <label className="quotation-field">
                Societ√†* (ragione sociale)
                <input type="text" id="company" name="company"/>
            </label>
            <label className="quotation-field">
                Indirizzo*
                <input type="text" id="address" name="address"/>
            </label>
            <label className="quotation-field">
                Posizione
                <input type="text" id="role" name="role"/>
            </label>
            <label className="quotation-field">
                e-mail*
                <input type="email" id="email" name="email"/>
            </label>
            <label className="quotation-field">
                Telefono
                <input type="tel" id="tel" name="tel"/>
            </label>
            <label className="quotation-field">
                Sito
                <input type="text" id="website" name="website"/>
            </label>
            <fieldset className="quotation-field">
                Motivo della richiesta:
                <label>
                    <input type="radio" name="reason" id="peoject"/>
                    Elaborazione progetto
                </label>
                <label>
                    <input type="radio" name="reason" id="call"/>
                    Miglioria per gara d'appalto
                </label>
                <label>
                    <input type="radio" name="reason" id="other"/>
                    Altro
                </label>
            </fieldset>
            <label className="quotation-field">
                Eventuali informazioni aggiuntive:
                <textarea className="quotation-msg" id="msg" name="msg"></textarea>
            </label>

            <button className="ctaBtn" type="submit">Invia</button>
        </form>
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Form);
