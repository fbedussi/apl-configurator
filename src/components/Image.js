import React from 'react';

const Image = ({ imgName }) => {
    if (!(imgName && imgName.length) ) {
        return null;
    }

    return <img className="image" src={`images/${imgName}`}/>
}

export default Image;