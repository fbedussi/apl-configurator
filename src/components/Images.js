import React from 'react';

const Images = ({ sources }) => {
    if (!(sources && sources.length) ) {
        return null;
    }

    return <div class="imagesContainer">
            sources.map(src => <img className="image" src={`images/${src}`}/>)
        </div>
}

export default Images;