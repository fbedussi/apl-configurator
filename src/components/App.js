import React from 'react';
import { connect } from 'react-redux';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

import { getSteps, getStepsLeft, getCurrentNode, getBreadcrumbs, getLabels, getGoingBack, getAnswers } from '../selectors';
import { parseAnswer, goBack } from '../actions';

import Answers from './Answers';
import Title from './Title';
import Back from './Back';
import Image from './Image';
import Recap from './Recap';
import Quotation from './Quotation';

const mapStateToProps = (state) => ({
    labels: getLabels(state),
    steps: getSteps(state),
    stepsLeft: getStepsLeft(state),
    goingBack: getGoingBack(state),
    currentNode: getCurrentNode(state),
    breadcrumbs: getBreadcrumbs(state),
    answers: getAnswers(state)
});

const mapDispatchToProps = (dispatch) => ({
    setAnswer: (value) => dispatch(parseAnswer(value)),
    goBack: (currentNode) => dispatch(goBack(currentNode))
})

class App extends React.Component {
    render() {
        return (
            <div className="configuratorWrapper">
                <div className="progressWrapper">
                    <span className="progressLabel">{this.props.labels["configurator"]}</span>
                    <progress className="progressBar" value={this.props.steps - this.props.stepsLeft} max={this.props.steps}></progress>
                </div>
                <div className={`workarea ${this.props.goingBack? 'back' : ''}`}>
                    <ReactCSSTransitionGroup
                        transitionName="slider"
                        transitionEnterTimeout={500}
                        transitionLeaveTimeout={500}>
                        <div className="slide" key={this.props.currentNode.id}>
                            <Title currentNode={this.props.currentNode} />
                            <p className={this.props.currentNode.type}>{this.props.currentNode.text}</p>
                            <Image 
                                imgName={this.props.currentNode.image}
                            />
                            <Answers
                                id={this.props.currentNode.id}
                                answers={this.props.currentNode.answers}
                                setAnswer={this.props.setAnswer}
                            />
                            <Back
                                show={Boolean(this.props.breadcrumbs.length)}
                                label={this.props.labels["back"]}
                                clickHandler={() =>
                                    this.props.goBack(this.props.currentNode)
                                }
                            />
                            <Recap
                                show={this.props.currentNode.type === 'answer'}
                                breadcrumbs={this.props.breadcrumbs}
                                answers={this.props.answers}
                            />
                            <Quotation
                                show={this.props.currentNode.type === 'answer'}
                            />
                        </div>

                    </ReactCSSTransitionGroup>

                </div>
            </div>
        );
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(App);