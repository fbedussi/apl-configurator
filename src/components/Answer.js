import React from 'react';
import { connect } from 'react-redux';
// import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

import { getLabels, getBreadcrumbs, getAnswersHistory, getShowForm, getAnswers, getQuestions } from '../selectors';
import { goBack, toggleForm } from '../actions';

import Title from './Title';
import Back from './Back';
import Images from './Images';
import Recap from './Recap';
import QuotationButton from './QuotationButton';
import Form from './Form';

const mapStateToProps = (state) => ({
    labels: getLabels(state),
    //currentNode: getCurrentNode(state),
    breadcrumbs: getBreadcrumbs(state),
    answersHistory: getAnswersHistory(state),
    answers: getAnswers(state),
    questions: getQuestions(state),
    showForm: getShowForm(state),
    //answer: getCurrentAnswer(state)
});

const mapDispatchToProps = (dispatch) => ({
    goBack: (currentNode) => dispatch(goBack(currentNode)),
    toggleForm: () => dispatch(toggleForm())
});

class Answer extends React.Component {
    render() {
        var answer = this.props.answers.filter(answer =>
            answer.id === this.props.currentNode.answerId)[0];

        return (
            <div className="slideInner">
                <Title text={answer.title} />
                {answer.subtitle && <p className="subtitle">{answer.subtitle}</p>}
                <p className="answer-text" dangerouslySetInnerHTML={{__html: answer.text}}/>
                <div className="answer-recapAndImage">
                    <Images
                        sources={answer.images}
                    />
                    <Recap
                        questions={this.props.questions}
                        labels={this.props.labels}
                        breadcrumbs={this.props.breadcrumbs}
                        answersHistory={this.props.answersHistory}
                    />
                </div>
                <Form
                    //showForm={this.props.showForm}
                    //solutionName={answer.title + ' - ' + answer.subtitle}
                />

                <QuotationButton
                    show={!this.props.showForm}
                    text={this.props.labels.quotationBtn}
                    clickHandler={() => this.props.toggleForm()}
                />

                <Back
                    show={Boolean(this.props.breadcrumbs.length)}
                    label={this.props.labels.back}
                    clickHandler={() =>
                        this.props.goBack(this.props.currentNode)
                    }
                />
            </div>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Answer);
