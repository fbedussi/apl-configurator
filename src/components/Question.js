import React from 'react';
import { connect } from 'react-redux';

import { getCurrentNode, getBreadcrumbs, getLabels, getGoingBack, getQuestions } from '../selectors';
import { parseAnswer, goBack } from '../actions';

import Title from './Title';
import Images from './Images';
import Options from './Options';
import Back from './Back';

const mapStateToProps = (state) => ({
    labels: getLabels(state),
    currentNode: getCurrentNode(state),
    questions: getQuestions(state),
    breadcrumbs: getBreadcrumbs(state)
});

const mapDispatchToProps = (dispatch) => ({
    setAnswer: (value) => dispatch(parseAnswer(value)),
    goBack: (currentNode) => dispatch(goBack(currentNode)),
})

export class Question extends React.Component {
    render() {
        var nodeId = isNaN(this.props.nodeId)? 1: this.props.nodeId;
        var question = this.props.questions.filter(question => question.id === nodeId)[0];
        
        return (
            <div className="slideInner">
                <Title currentNode={question.title} />
                <p className="text question-text" dangerouslySetInnerHTML={{__html: question.text}} />

                <Images
                    sources={question.images}
                />
                <Options
                    currentNode={this.props.currentNode}
                    options={question.answers}
                    setAnswer={this.props.setAnswer}
                    baseUrl={this.props.baseUrl}
                    history={this.props.history}
                />
                <Back
                    show={Boolean(this.props.breadcrumbs.length)}
                    label={this.props.labels["back"]}
                    clickHandler={() =>
                        this.props.goBack(this.props.currentNode)
                    }
                />
            </div>
        );
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(Question);