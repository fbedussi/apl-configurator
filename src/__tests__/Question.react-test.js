import React from 'react';

import {Question} from '../components/Question';
import renderer from 'react-test-renderer';
import texts from './apl-texts.json'

var labels = texts.labels;
var questions = texts.questions;
var currentNode = texts.tree;
var breadcrumbs = [];

test('Question component', () => {
  const component = renderer.create(
    <Question labels={labels} questions={questions} currentNode={currentNode} breadcrumbs={breadcrumbs} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});