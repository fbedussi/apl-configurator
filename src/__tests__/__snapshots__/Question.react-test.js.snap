// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component 1`] = `
<div
  className="slideInner"
>
  <p
    className="text question-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Nel luogo di installazione è possibile collegare il sistema APL alla rete elettrica?",
      }
    }
  />
  <div
    className="question-options"
  >
    <button
      className="question-optionBtn ctaBtn"
      onClick={[Function]}
    >
      no
    </button>
    <button
      className="question-optionBtn ctaBtn"
      onClick={[Function]}
    >
      sì
    </button>
  </div>
</div>
`;
